assets/js/modules/ads/components/common/WooCommerceRedirectModal.js

```diff
@@ -229,11 +245,25 @@ export default function WooCommerceRedirectModal( {
 							<CircularProgress size={ 14 } />
 						) : undefined
 					}
-					onClick={
-						isGoogleForWooCommerceAdsConnected
-							? getGoogleForWooCommerceRedirectURI
-							: onClose
-					}
+					onClick={ () => {
+						if (
+							( isWooCommerceActive ||
+								isGoogleForWooCommerceAdsConnected ) &&
+							! isAccountLinkedViaGoogleForWoocommerceNoticeDismissed
+						) {
+							dismissNotification(
+								'account-linked-via-google-for-woocommerce'
+							);
+						}
```

Comment:
Do we really need to check whether the WooCommerce is active and account is not linked? The ticket says that we should dismiss this notification whenever the user is redirected to the WooCommerce redirection URL. This means we should dismiss this notification in the callback function right before navigating, right?

Reply:
@eugene-manuilov it is not checking if account is not linked, it is checking if account not linked _notice_ is dismissed already before dismissing it. It is much of a muchness, but moving it to the redirect logic may be a better place, sure. I will make that change.

Reply:
Agreed that a callback is a better place, I have repurposed `getGoogleForWooCommerceRedirectURL` to be called `handleGoogleForWooCommerceRedirect` instead, and bundled logic therein. Thanks @eugene-manuilov